demisto_sdk/commands/validate/validators/RN_validators/RN114_validate_release_notes_header.py

```diff
@@ -0,0 +1,255 @@
+                validator_results.append(
+                    ValidationResult(
+                        validator=self,
+                        message=self.error_message.format(
+                            content_type_message=content_type_message,
+                            content_item_message=content_item_message,
+                        ),
+                        content_object=content_item,
+                    )
+                )
+        return validator_results
+
+    def remove_none_values(self, ls: Union[List[Any], Tuple[Any, ...]]) -> List[Any]:
+        """
+        Filters out None values from a list or tuple.
+
+        Args:
+            ls (List or Tuple): The list or tuple to filter.
+
+        Returns:
+            List: Filtered list with None values removed.
+        """
+        return list(filter(None, ls))
+
+    def extract_rn_headers(self, release_note_content: str) -> Dict[str, List[str]]:
+        """
+        Extracts the headers from the release notes file.
+
+        Args:
+            release_note_content (str): Content of the release notes file.
+
+        Return:
+            Dict[str, List[str]]: A dictionary representation of the release notes file that maps
+                                  content types' headers to their corresponding content items' headers.
+        """
+        headers: Dict[str, List[str]] = {}
+        content_type_section_pattern = re.compile(CONTENT_TYPE_SECTION_REGEX, re.M)
+        content_item_section_pattern = re.compile(CONTENT_ITEM_SECTION_REGEX, re.M)
+
+        # Get all sections from the release notes using regex
+        rn_sections = content_type_section_pattern.findall(release_note_content)
+        for section in rn_sections:
+            section = self.remove_none_values(ls=section)
+            if not section:
+                continue
+
+            content_type = section[0]
+            content_type_sections_str = section[1]
+            content_type_sections_ls = content_item_section_pattern.findall(
+                content_type_sections_str
+            )
+
+            if not content_type_sections_ls:
+                # Did not find content items headers under content type - might be due to invalid format.
+                # Will raise error in rn_valid_header_format.
+                headers[content_type] = []
+
+            for content_type_section in content_type_sections_ls:
+                content_type_section = self.remove_none_values(ls=content_type_section)
+                if content_type_section:
+                    header = (
+                        content_type_section[0].rstrip()
+                        if ("New: " not in content_type_section[0])
+                        else content_type_section[0][5:]
+                    )
+                    if content_type in headers:
+                        headers[content_type].append(header)
+                    else:
+                        headers[content_type] = [header]
```

Comment:
this can be much clearer using `headers = defaultdict(list)`, and then always appending (even if the key doesn't exist, defaultdict handles it).